src/celestia/qt/qtappwin.cpp

```diff
@@ -861,6 +862,14 @@ CelestiaAppWindow::gotoObject()
 }
 
 
+void
+CelestiaAppWindow::tourGuide()
+{
+    TourGuideDialog* tourDialog = new TourGuideDialog(this, m_appCore);
```

Comment:
resource leaking here. `TourGuideDialog *tourDialog` should be part of `CelestiaAppWindow`.
so this function will be:
```c++
if (tourDialog == nullptr)
    tourDialog = new TourGuideDialog(this, m_appCore);

tourDialog->show();
```

Reply:
**`If`** code displays second pass dialog (window->close->window) at display point it was on first window, not at the beginning.  Did not add this code.

All dialog buttons now set to emit close() event.  Qt attribute set to destroy dialog on close event.  Still allows multiple windows, but does appear to delete the dialog window.  Dialog windows start at the beginning.  This change should stop resource leak.  Submitted this code.

Tried both together.  First pass window successful.  Second pass window crashes system.  I think this proves the window deletion - the pointer and associated dialog information is deleted upon first window closure and when `if` statement is activated no tourDialog pointer causes crash.

Added comments that window is special needing non-modal opening to prevent confusion in future.

Original code is found in other GTK and Qt menu items - should leak there.  Assume is also in win32.

Reply:
I see you set `Qt::WA_DeleteOnClose` in the `TourGuideDialog` constructor so I think this should be ok?